src/config/cli_config.cpp

```diff
@@ -0,0 +1,145 @@
+                                     "JSON configuration file (default: \"~/.taraxa/config.json\")");
+  node_command_options.add_options()(DATA_DIR, bpo::value<string>(&data_dir),
+                                     "Data directory for the databases (default: \"~/.taraxa/db\")");
+  node_command_options.add_options()(DESTROY_DB, bpo::bool_switch(&destroy_db),
+                                     "Destroys all the existing data in the database");
+  node_command_options.add_options()(REBUILD_DB, bpo::bool_switch(&rebuild_db),
+                                     "Reads the raw dag/pbft blocks from the db "
+                                     "and executes all the blocks from scratch "
+                                     "rebuilding all the other "
+                                     "database tables - this could take a long "
+                                     "time");
+  node_command_options.add_options()(REBUILD_DB_PERIOD, bpo::value<uint64_t>(&rebuild_db_period),
+                                     "Use with rebuild-db - Rebuild db up "
+                                     "to a specified period");
+  node_command_options.add_options()(REBUILD_NETWORK, bpo::bool_switch(&rebuild_network),
+                                     "Delete all saved network/nodes information "
+                                     "and rebuild network from boot nodes");
+  node_command_options.add_options()(REVERT_TO_PERIOD, bpo::value<uint64_t>(&revert_to_period),
+                                     "Revert db/state to specified "
+                                     "period (specify period)");
+  node_command_options.add_options()(NETWORK_ID, bpo::value<int>(&network_id),
+                                     "Network identifier (integer, 1=Mainnet, 2=Testnet, 3=Devnet) (default: 2)");
+  node_command_options.add_options()(BOOT_NODE, bpo::bool_switch(&boot_node), "Run as bootnode (default: false)");
+
+  allowed_options.add(main_options);
+  allowed_options.add(node_command_options);
+  bpo::variables_map option_vars;
+  auto parsed_line = bpo::parse_command_line(argc, argv, allowed_options);
+  bpo::store(parsed_line, option_vars);
+  bpo::notify(option_vars);
+  if (option_vars.count(HELP)) {
+    cout << "NAME:\n  "
+            "taraxad - Taraxa blockchain full node implementation\n"
+            "VERSION:\n  "
+         << taraxa_version << "\nUSAGE:\n  taraxad [options]\n";
+    cout << main_options << endl;
+    cout << node_command_options << endl;
+    // If help message requested, ignore any additional commands
+    command.clear();
+    return;
+  }
+  if (version) {
+    cout << taraxa_version << endl;
+    // If version requested, ignore any additional commands
+    command.clear();
+    return;
+  }
+
+  // Create dir if missing
+  if (!fs::exists(data_dir)) {
+    fs::create_directory(data_dir);
+  }
+  if (!fs::exists(default_dir)) {
+    fs::create_directory(default_dir);
+  }
```

Comment:
With default options this should fail as line 90 does not create the parent directory `.taraxa`  and the default `data_dir` is `.taraxa/db` 
Changing the order may solve the issue with default values.

I am not sure if we want to create parent dirs automatically when passed as parameters, if we do, this won't work as well, it would need to check all parents and create them as needed.